### DISCUSSION TITLE
请问下如何自定义blockID？

### DISCUSSION THREAD

**Discussion by CoLin00** [aidenlx/obsidian-zotero#216] [Help]
您好，我需要自定义blockID的编写规则，请问下应该如何设定模板？

`zt-annot.eta`模板里面设置的都会被包括在引用“>"之中，也就会被放进annotation的模版中。我可否在模版中设定，实现：

1. blockID 在annotation之外，即位于annotation下面。
2. blockID可以自己按特定规则编写，如以每个引用中的Key为设定方式，以缩短blockID，或按其他要求进行方便的检索。

**Comment by CoLin00** [comment]
如果可以将默认添加blockID关掉也行的，但是不知道如何关闭，求大佬支招。

**Reply by aidenlx** [reply]
这边有写怎么禁用，不好意思文档内容还是有点分散

https://zotlit.aidenlx.top/getting-started/basic-usage/update-notes#full-update

**Reply by CoLin00** [reply]
感谢您，能够去掉默认的blockid，但是在这种情况下还存在一些小问题。就是只能在关闭每行添加>的情况下取消掉blockid，但是会出现我下面所说的情况。当我以手动的方式在模版中每行前添加>的时候，`<%= it.comment %> `部分的批注是分了多行的时候，我只能让第一行的批注呈现callout样式。不知道是否可以单独设置一个blockid的yaml元素，这样可能会比较好吧？

**Reply by CoLin00** [reply]
不过，我用了相对折中的方式，在最后添加`---`，用线把每一块区分开来了。

**Comment by CoLin00** [comment]
如果我设定了：
```
---
callout: false
---

```
并且，我手动在所有内容前加上“>”，虽然默认的blockid没有了，但是当我的comment是多行的时候，">" 只能输出一个，其他行的就会没有添加上，使得整个内容无法被框进>里面去。

**Comment by aidenlx** [comment]
v2 不再往标注里写 block ID，所以这个问题本身不存在了。

更新靠的是 `%%zt-managed%%` 标记圈出的托管区域整体重新渲染，不需要用 block ID 去定位单条标注，标记之外的内容不会被动到。引用块（`>`）也只是默认 annotation 模板里的一个 `{% bq %}` 标签，删掉就没有了 —— 而且它会正确处理多行 comment 的每一行前缀，不会再出现只有第一行加上 `>` 的情况。

相关说明：https://zotlit.aidenlx.site/docs/concepts/literature-notes-and-the-managed-region

v2 beta 现已发布：https://zotlit.aidenlx.site/docs/install-zotlit
迁移指南：https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
